From 21dc49cb8ce4ee5fb19ae94eacb802a55cc1c25d Mon Sep 17 00:00:00 2001
From: Ferdinand Xu <cheng.a.xu@intel.com>
Date: Thu, 24 Sep 2015 21:14:33 -0400
Subject: [PATCH 0716/1164] CDH-42377 : HIVE-11943: Set old CLI as the default
 Client when using hive script(Ferdinand Xu,
 reviewed by Xuefu Zhang)

Change-Id: I9a4fd3da15a01db0e90c59d805c792eefc9b8e18
---
 bin/ext/cli.cmd |    6 +++++-
 bin/ext/cli.sh  |    6 +++++-
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/bin/ext/cli.cmd b/bin/ext/cli.cmd
index 3a96d03..d9185c0 100644
--- a/bin/ext/cli.cmd
+++ b/bin/ext/cli.cmd
@@ -28,9 +28,13 @@ goto :EOF
 goto :EOF
 
 :update_cli
+	if [%USE_DEPRECATED_CLI%] == [] (
+		set USE_DEPRECATED_CLI=false
+	)
+
 	if /I "%USE_DEPRECATED_CLI%" == "true" (
 		call :old_cli
-	) else if (
+	) else (
 		call :new_cli
 	)
 goto :EOF
diff --git a/bin/ext/cli.sh b/bin/ext/cli.sh
index 7b9c181..353f297 100644
--- a/bin/ext/cli.sh
+++ b/bin/ext/cli.sh
@@ -16,10 +16,14 @@
 THISSERVICE=cli
 export SERVICE_LIST="${SERVICE_LIST}${THISSERVICE} "
 
+# Set old CLI as the default client
+if [ -n '$USE_DEPRECATED_CLI' ]; then
+  USE_DEPRECATED_CLI="true"
+fi
+
 updateBeelineOpts() {
   # If process is backgrounded, don't change terminal settings
   if [[ ! $(ps -o stat= -p $$) == *+ ]]; then
-    echo "background"
     export HADOOP_CLIENT_OPTS="$HADOOP_CLIENT_OPTS -Djline.terminal=jline.UnsupportedTerminal"
   fi
 }
-- 
1.7.9.5

